contracts/insured-vesting-v1/InsuredVestingV1.sol

```diff
@@ -63,12 +64,19 @@ contract InsuredVestingV1 is Ownable {
     error NoFundsAdded();
     error EmergencyReleased();
     error EmergencyNotReleased();
+    error AlreadyEmergencyReleased();
+    error OnlyOwnerOrSender();
 
     modifier onlyBeforeVesting() {
         if (startTime != 0 && block.timestamp > startTime) revert VestingAlreadyStarted();
         _;
     }
 
+    modifier onlyOwnerOrSender(address target) {
+        if (msg.sender != owner() && msg.sender != target) revert OnlyOwnerOrSender();
```

Comment:
I think this is wrong, no? You want to enforce the user is **either** the owner **or** an investor, but you are currently enforcing they are both (`&&`)?

Reply:
the NOT is what makes the change - (if you're not ALLOWED_USER_1 AND you're not ALLOWED_USER_2) then you're not allowed

Reply:
this could be rewritten as `if !(msg.sender == owner() || msg.sender == target)`